pkg/controller/windowsmachineconfig/nodeconfig/nodeconfig.go

```diff
@@ -48,9 +49,39 @@ func (nc *nodeConfig) Configure() error {
 	if err := nc.handleCSRs(); err != nil {
 		return errors.Wrap(err, "handling CSR for the given node failed")
 	}
+
+	// Now that basic kubelet configuration is complete, configure networking in the node
+	if err := nc.configureNetwork(); err != nil {
+		return errors.Wrap(err, "configuring networking node network failed")
+	}
+	log.Info("VM has been configured as a worker node", "VM ID", nc.GetCredentials().GetInstanceId())
 	return nil
 }
 
+// configureNetwork configures k8s networking in the node
+func (nc *nodeConfig) configureNetwork() error {
+	if nc.GetCredentials() == nil {
+		return fmt.Errorf("nil credentials for VM")
+	}
+
+	if nc.GetCredentials().GetIPAddress() == "" {
+		return fmt.Errorf("empty external IP for VM: %v", nc.GetCredentials())
```

Comment:
I dont think we should mention external IP, as we might change the `GetIPAddress()` implementation to return the private IP